### PR TITLE
Fix newError noops when not bound to dispacher

### PR DESCRIPTION
**Pull request by jacogr**
newError in AddContract, EditMeta & PasswordManager was not properly bound.

### CODE REVIEW

File: js/src/modals/AddContract/addContract.js [hunk 1]
```diff
@@ -244,11 +247,26 @@ export default class AddContract extends Component {
...
+}
+
+function mapDispatchToProps (dispatch) {
+  return bindActionCreators({
+    newError
+  }, dispatch);
+}
+
+export default connect(
+  mapStateToProps,
```
Comment (derhuerst) [inline js/src/modals/AddContract/addContract.js]: This is just an optimisation:

> If you omit [`mapDispatchToProps`], the component will not be subscribed to the Redux store.

You could pass `null`, it occurs multiple times in the codebase already.

File: js/src/modals/AddContract/addContract.spec.js [hunk 2]
```diff
@@ -20,36 +20,65 @@ import sinon from 'sinon';
...
     });
 
     it('renders the defauls', () => {
       expect(component).to.be.ok;
     });
   });
+
+  describe('onAdd', () => {
+    it('calls store addContract', () => {
+      sinon.stub(instance.store, 'addContract').resolves(true);
```
Comment (derhuerst) [inline js/src/modals/AddContract/addContract.spec.js]: I think in this case it's not worth it yet, but once we have more stubs, restoring them can easily be forgotten (which breaks the whole point of the stubs then). [sandboxes](http://sinonjs.org/docs/#sinon-sandbox) try to address this issue.

File: js/src/modals/AddContract/addContract.js (hunk 1 above)
Comment (jacogr) [inline js/src/modals/AddContract/addContract.js]: I generally prefer being explicit and in a once-off situation being explicit about an empty state is not bad. However who knows what happens with Redux making that call each cycle as things changes, it just add unnecessary overheads. 

Will update.

File: js/src/modals/EditMeta/editMeta.js [hunk 3]
```diff
@@ -138,6 +142,24 @@ export default class EditMeta extends Component {
 
     return this.store
       .save()
-      .then(() => this.props.onClose());
+      .then(() => this.props.onClose())
+      .catch((error) => {
```
Comment (derhuerst) [inline js/src/modals/EditMeta/editMeta.js]: 👍

File: js/src/modals/EditMeta/editMeta.js [hunk 4]
```diff
@@ -138,6 +142,24 @@ export default class EditMeta extends Component {
...
+}
+
+function mapDispatchToProps (dispatch) {
+  return bindActionCreators({
+    newError
+  }, dispatch);
+}
+
+export default connect(
+  mapStateToProps,
```
Comment (derhuerst) [inline js/src/modals/EditMeta/editMeta.js]: As above.

**Pull request merged**